gci/oci.py

```diff
@@ -103,3 +106,26 @@ def component_descriptor_from_tarfileobj(
           raise ValueError('Component Descriptor appears to be empty')
 
         return gci.componentmodel.ComponentDescriptor.from_dict(raw_dict)
+
+
+def image_ref_with_digest(
```

Comment:
I think this function might better fit into oci.util (w/o DigestSpec, of course) - wdyt?

Reply:
If you move it to `oci.util`, there would be some obstacles because of circular imports. I'd suggest to keep it in `gci/oci` since it contains both OCM and OCI related aspects 🤔